src/Raven.Server/ServerWide/ClusterStateMachine.License.cs

```diff
@@ -391,72 +393,149 @@ void AssertAutoIndexesCount()
+            return;
+
+        var includeRevisions = putSubscriptionCommand.Query.Contains(DocumentSubscriptions.IncludeRevisionsRQL);
+        AssertSubscriptionRevisionFeatureLimits(serverStore, includeRevisions, context);
+    }
+
+    private List<T> AssertSubscriptionsBatchLicenseLimits<T>(ServerStore serverStore, Table items, BlittableJsonReaderArray subscriptionCommands, string type, ClusterOperationContext context)
+        where T: PutSubscriptionCommand
+    {
+        var includeRevisions = false;
+        var putSubscriptionCommandsList = new List<T>();
+
+        foreach (BlittableJsonReaderObject command in subscriptionCommands)
         {
-            var clusterSubscriptionsCounts = GetSubscriptionsCount();
-            if (clusterSubscriptionsCounts + 1 > maxSubscriptionsPerCluster)
-            {
-                if (CanAssertLicenseLimits(context, minBuildVersion: MinBuildVersion60000) == false)
-                    return;
+            if (command.TryGet("Type", out string putSubscriptionType) == false && putSubscriptionType != nameof(T))
+                throw new RachisApplyException($"Cannot execute {type} command, wrong format");
 
-                throw new LicenseLimitException(LimitType.Subscriptions, $"The maximum number of subscriptions per cluster cannot exceed the limit of: {maxSubscriptionsPerCluster}");
-            }
+            var putSubscriptionCommand = (T)JsonDeserializationCluster.Commands[typeof(T).Name](command);
+            putSubscriptionCommandsList.Add(putSubscriptionCommand);
+
+            if (includeRevisions == false &&
+                putSubscriptionCommand.Query.Contains(DocumentSubscriptions.IncludeRevisionsRQL))
+                includeRevisions = true;
         }
 
-        if (serverStore.LicenseManager.LicenseStatus.HasRevisionsInSubscriptions == false &&
-            command.Query.Contains(DocumentSubscriptions.IncludeRevisionsRQL))
+        if (AssertSubscriptionRevisionFeatureLimits(serverStore, includeRevisions, context) == false)
+            return putSubscriptionCommandsList;
+
+        var subscriptionsNamesPerDatabase = new Dictionary<string, List<string>>();
+        foreach (var command in putSubscriptionCommandsList)
         {
-            if (CanAssertLicenseLimits(context, minBuildVersion: MinBuildVersion60000) == false)
-                return;
+            if (subscriptionsNamesPerDatabase.TryGetValue(command.DatabaseName, out _) == false)
+                subscriptionsNamesPerDatabase.Add(command.DatabaseName, new List<string>());
 
-            throw new LicenseLimitException(LimitType.Subscriptions, "Your license doesn't include the subscription revisions feature.");
+            subscriptionsNamesPerDatabase[command.DatabaseName].Add(command.SubscriptionName);
         }
 
-        long GetSubscriptionsCount()
+        if (AssertNumberOfSubscriptionsPerClusterLimits(serverStore, items, context, subscriptionsNamesPerDatabase))
```

Comment:
lets keep same order as in first methd, first database limits then cluster limits

Reply:
Done.